### PR TITLE
Color by HP tags

### PR DESCRIPTION
**Pull request by MariaNattestad**
Screenshot 2024-08-26 at 4 52 32 PM

### CODE REVIEW

File: js/index_ribbon.js [hunk 1]
```diff
@@ -1566,24 +1577,59 @@ function draw_chunk_alignments() {
...
-    if (_ribbon_settings.orient_reads_by == "primary") {
-      chunks[i].flip =
-        chunks[i].unfiltered_alignments[chunks[i].index_primary].qe -
-          chunks[i].unfiltered_alignments[chunks[i].index_primary].qs <
-        0;
-    } else if (_ribbon_settings.orient_reads_by == "longest") {
-      chunks[i].flip =
-        chunks[i].unfiltered_alignments[chunks[i].index_longest].qe -
-          chunks[i].unfiltered_alignments[chunks[i].index_longest].qs <
-        0;
```
Comment (MariaNattestad) [inline js/index_ribbon.js:1582]: I refactored this repetitive code for clarity while I was in the neighborhood.

Comment (robertaboukhalil) [reply js/index_ribbon.js:1582]: https://www.youtube.com/watch?v=FhAJnx5uwUU

**Review by robertaboukhalil — APPROVED** [review]
Very nice!

File: js/index_ribbon.js [hunk 2]
```diff
@@ -1658,21 +1704,7 @@ function draw_chunk_alignments() {
...
-            return "red";
-          }
-        } else {
-          if (d.qs < d.qe) {
-            return "red";
-          } else {
-            return "blue";
-          }
-        }
-      })
```
Comment (MariaNattestad) [inline js/index_ribbon.js:1675]: I'm slightly horrified that I ever wrote this code. Now the color is computed up above and just attached to the `d` object here, which represents an alignment.

Comment (MariaNattestad) [reply js/index_ribbon.js:1675]: (let alone did it four times...)

File: js/index_ribbon.js (hunk 1 above)
Comment (MariaNattestad) [reply js/index_ribbon.js:1582]: It's a beautiful day in this neighborhood :)

**Pull request merged**